staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -952,12 +976,20 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 		}
 
 		structural, err := structuralschema.NewStructural(schema)
+		celCRDCost := uint64(0)
 		if err == nil {
 			compResults, err := cel.Compile(structural, isRoot, cel.PerCallLimit)
 			if err != nil {
 				allErrs = append(allErrs, field.InternalError(fldPath.Child("x-kubernetes-validations"), err))
 			} else {
 				for i, cr := range compResults {
+					celCRDCost += getExpressionCost(cr.MaxCost, nodeCostInfo)
+					if celCRDCost > TotalCostLimit {
+						// TODO(DangerOnTheRanger): consider how to make the error message more informative
+						exceedFactor := float64(celCRDCost) / float64(TotalCostLimit)
+						costErrorMsg := fmt.Sprintf("CEL rule exceeded budget by factor of %vx", exceedFactor)
```

Comment:
Maybe put "cost" in the error and add some details about the budget? E.g.

`CEL rule cost of %d exceeded budget of %d by factor of %vx`

?

Reply:
Yes, that's a lot better. I've removed the TODO and am marking this as resolved